### DISCUSSION TITLE
Set window open mode to thermostate

### DISCUSSION THREAD

**Discussion by HardwareToWebWizard** [SukramJ/aiohomematic#658] [Ideas]
Hi, is there a way to set the window open mode on the thermostate? I want to save the state of the thermostate and restore it after the window is being closed again. I use other hardware than homematic for window open detection so it would be nice if i can send this command to my thermostates.

kind regards

**Comment by SukramJ** [comment]
There is a community [blueprint](https://community.home-assistant.io/t/homematic-ip-local-reduce-room-temperature-when-a-window-or-door-is-opened/373305) to might help. This one also includes the requested service call.

**Comment by HardwareToWebWizard** [comment]
Thank you very much i will have a look

**Comment by HardwareToWebWizard** [comment]
Should this also work for Homematic devices which are not IP ? I have older homematic devices and not the new ip ones and this blueprint seems not working...

**Reply by SukramJ** [reply]
A device_type is might be useful.

**Reply by SukramJ** [reply]
I am not sure if there is any HM climate device, that supports a window state.
Maybe others know better.

**Comment by Baxxy13** [comment]
The used Datapoint "WINDOW_STATE" is only available within IP-Wall/Radiator Thermostats. 
Old HM-Devices dont have this.
As example you could set them to MANU / 10°C when window opens and AUTO if window closes.

**Reply by HardwareToWebWizard** [reply]
Well that is exactly not what i want as i want to set the old state of the thermostat once the window is closed again. So i thought with the window open detection/mode it would be easier to set the state before the window was opened

**Reply by ThomDietrich** [reply]
Hey,
this might not be the solution you are looking for but my working system works like this:

- One helper entity holds the set temperature for a thermostat, ignorning window states. This entity is manipulated by dozens of automations to account for e.g. a time program, house-wide modes like "vacation", or my calendar. Manual changes are also possible.
- One helper entity to define when "lowered mode" should be active (e.g. when window is open or the kitchen ventilation is running)
- Automations that set the thermostat based on the above set temperature and under consideration of the "lowered mode" binary. This includes "i want to set the old state of the thermostat once the window is closed again."
- HM thermostats are permanently in MANU mode. All power to HA! :D

So as you can see, I didn't even bother to look into the window mode function of the thermostat - leaving me with more HA flexibility. Let me know if you would like to see and potentially copy my solution yaml.

**Reply by HardwareToWebWizard** [reply]
I really would like to see your solution. Seems like this is a very good workaround :D

**Reply by stuckis6er** [reply]
Hallo,
meine Idee wäre es das mit Scenen zu realisieren. Also wenn sich das Fenster öffnet den aktuellen Zustand des Thermostat in eine Scene speichern und dann auf Manuell XX Grad einstellen. Wenn das Fenster wieder zu ist die gespeicherte Scene wieder aktivieren.

Gruß Ralf

**Reply by ThomDietrich** [reply]
Hey @PhilThal,
I've uploaded my complete heating solution here: https://gist.github.com/ThomDietrich/1151299494432f2503cf9b00187aaf0d

Almost everything is commented in german, hope that's useful for you.

You should be able to easily adapt this to your home, you just need to edit and potentially duplicate everything related to rooms. So e.g. instead of my `binary_sensor.bad_heizstellglied_lower` you might need `binary_sensor.office_thermostat_lower`.

Very much looking forward to any kind of questions and improvement suggestions ;)

**Reply by HardwareToWebWizard** [reply]
Thanks, i am german but wanted to diskuss this topic for everyone :D I will look into your script. Thank you for sharing

**Reply by HardwareToWebWizard** [reply]
Seems like i got it to work now with at least one thermostat. The other ones are still bit buggy... best solution for me would still be the option to set the window open mode. Alone the amount of code needed for this in automations etc ... would be easier to just set it within the homematic ip addon .....

**Reply by Lemocin** [reply]
Ich habe dafür eine Sysvar je Raum:
![grafik](https://user-images.githubusercontent.com/4162496/199097432-edae6bff-f127-4f1d-a4bb-765546a34731.png)
und das dazugehörige Programm:
![grafik](https://user-images.githubusercontent.com/4162496/199097527-c2370d1b-4ede-474f-86af-c78d8ae2ac02.png)

Dazu dann die Automatisierung in HA:
![grafik](https://user-images.githubusercontent.com/4162496/199097826-3dd71fd9-1118-4de0-acd2-aa5cce028ffe.png)
![grafik](https://user-images.githubusercontent.com/4162496/199097857-b2d8d87e-4ccf-4b80-bb91-d983ef6dc307.png)


Funktioniert bei mir schon seit Jahren super!

**Reply by HardwareToWebWizard** [reply]
danke für eure hilfe :D @ThomDietrich jetzt hab ich auch noch meinen letzten fehler gefunden und es funktioniert. Das mit der Sysvar schaue ich mir auch nochmal an das sieht auch sehr interessant aus. Nochmals vielen Dank fürs teilen von euren Lösungen!
